src/nsrt_learning.py

```diff
@@ -75,13 +77,25 @@ def segment_trajectory(trajectory: GroundAtomTrajectory) -> List[Segment]:
     for t in range(len(actions)):
         current_segment_traj[0].append(states[t])
         current_segment_traj[1].append(actions[t])
-        if all_atoms[t] != all_atoms[t+1]:
+        switch = all_atoms[t] != all_atoms[t+1]
+        if actions[t].has_option() and t < len(actions) - 1:
```

Comment:
This is fixing an otherwise deep bug. Let's include a detailed comment about it.